File: WorkSpace/test/bin/ch07/ch7_08.cs
//檔名:ch7_08.cs   功能:傳參考呼叫

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ConsoleApplication1
{
    class Program
    {
        static void Main(string[] args)
        {
            CVector2 i = new CVector2();
            i.set(20, 40);
            CVector2 j = new CVector2();
            j.set(15, 45);
            Console.WriteLine("-----原始------");
            Console.WriteLine("Vector i=(" + i.x + "," + i.y + ")");
            Console.WriteLine("Vector j=(" + j.x + "," + j.y + ")");
            CMyClass X = new CMyClass();
            X.swap(ref i, ref j);   //傳參考呼叫
            Console.WriteLine("-----swap後-------");
            Console.WriteLine("Vector i=(" + i.x + "," + i.y + ")");
            Console.WriteLine("Vector j=(" + j.x + "," + j.y + ")");
            Console.Read();
        }
    }
    class CMyClass
    {
        public void swap(ref CVector2 a, ref CVector2 b)  //傳參考呼叫的互換函式
        {
            CVector2 temp;
            temp = a;
            a = b;
            b = temp;
        }
    }

    class CVector2                      //二維向量類別
    {
        public int x, y;               //二維向量的兩項元素資料
        public void set(int m, int n)  //用於設定二維向量的兩項元素資料
        {
            x = m;
            y = n;
        }
    }
}

File: WorkSpace/test/bin/ch07/ch7_08.cpp
/*   檔名:ch7_08.cpp   功能:傳參考呼叫   */

#include <iostream>
#include <stdlib.h>
using namespace std;

class CVector2                      //二維向量類別
{
  public: int x,y;               //二維向量的兩項元素資料
  public: void set(int m,int n)  //用於設定二維向量的兩項元素資料
  {          
        x=m;
        y=n;      
  }
};

class CMyClass
{
    public: void swap(CVector2 &a,CVector2 &b)  //互換函式,傳參考呼叫
    {
        CVector2 temp;
        temp=a;
        a=b;
        b=temp;
    }
};

int main(char args[])
{
   CVector2 i;
   i.set(20,40);
   CVector2 j;
   j.set(15,45);
   cout << "-----原始------" << endl;      //列印
   cout << "Vector i=(" << i.x << "," << i.y << ")" << endl;
   cout << "Vector j=(" << j.x << "," << j.y << ")" << endl;
   CMyClass X;
   X.swap(i,j);   //互換i,j
   cout << "-----swap後-------" << endl;   //列印
   cout << "Vector i=(" << i.x << "," << i.y << ")" << endl;
   cout << "Vector j=(" << j.x << "," << j.y << ")" << endl;
   //system("pause");
   return 0;
}


